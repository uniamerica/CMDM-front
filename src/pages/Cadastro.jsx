import React, { useState } from 'react'
import Header from '../components/Header/Header'

import styles from '../styles/Cadastro.module.css'
import api from '../api/api'

export default function Cadstro() {
    const [newUser, setNewUser] = useState({})

    function handleChangeNome(props) {
        setNewUser({...newUser,
            name: props.target.value
        })
    }
    
    function handleChangeEmail(props) {
        setNewUser({...newUser,
            email: props.target.value
        })
    }
    
    function handleChangePhone(props) {
        setNewUser({...newUser,
            phone: props.target.value
        })
    }
    
    function handleChangeBirth(props) {
        setNewUser({...newUser,
            birth: props.target.value
        })
    }
    
    function handleChangeUsername(props) {
        setNewUser({...newUser,
            username: props.target.value
        })
    }
    
    function handleChangePassword(props) {
        setNewUser({...newUser,
            password: props.target.value
        })
    }

    function handleSubmit(event) {
        api.post('/users', newUser)
    }

    return (
        <>
            {/* <Header /> */}
            <div className={styles.formCadastro}>
                <form onSubmit={handleSubmit}>
                    <div className={styles.formField}>
                        <label>Nome completo</label>
                        <input id="name" type="text" name="name" placeholder="Maria da Penha" value={newUser.name} onChange={handleChangeNome}></input>
                    </div>
                    <div className={styles.formField}>
                        <label>Email</label>
                        <input id="mail" type="text" name="mail" placeholder="maria@email.com" value={newUser.email} onChange={handleChangeEmail}></input>
                    </div>
                    <div className={styles.formField}>
                        <label>Telefone</label>
                        <input id="phone" type="text" name="phone" placeholder="(DDD) 99999-0000" value={newUser.phone} onChange={handleChangePhone}></input>
                    </div>
                    <div className={styles.formField}>
                        <label>Data de nascimento</label>
                        <input id="birth" type="date" name="birth" placeholder="01/01/1990" value={newUser.birth} onChange={handleChangeBirth}></input>
                    </div>
                    <div className={styles.formField}>
                        <label>Nome de usuário</label>
                        <input id="username" type="text" name="username" placeholder="MariaPenha123" value={newUser.username} onChange={handleChangeUsername}></input>
                    </div>
                    <div className={styles.formField}>
                        <label>Senha</label>
                        <input id="password" type="text" name="password" placeholder="************" value={newUser.password} onChange={handleChangePassword}></input>
                    </div>
                    <button name="submitButton" id="cadastroSubmit" type="submit" className={styles.submit}>Cadastrar-se</button>
                    <a href="/login">
                        Já tem cadastro?
                    </a>
                </form>
            </div>
        </>
    )
}