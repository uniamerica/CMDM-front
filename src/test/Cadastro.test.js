import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import '@testing-library/jest-dom'
import Cadastro from '../pages/Cadastro'
import Header from '../components/Header/Header'
import { MemoryRouter } from 'react-router'

describe("Cadastro Component", () => {
    test("should verify if the component had changes", () => {
        const component = render(<Cadastro/>)

        expect(component).toMatchSnapshot();
    })

    test("should have inputs in the form", () => {
        render(<Cadastro/>)
        
        const nameInput = screen.getByPlaceholderText('Maria da Penha')
        const emailInput = screen.getByPlaceholderText('maria@email.com')
        const phoneInput = screen.getByPlaceholderText('(DDD) 99999-0000')
        const birthInput = screen.getByPlaceholderText('01/01/1990')
        const usernameInput = screen.getByPlaceholderText('MariaPenha123')
        const passwordInput = screen.getByPlaceholderText('************')
        expect(nameInput).toBeInTheDocument()
        expect(emailInput).toBeInTheDocument()
        expect(phoneInput).toBeInTheDocument()
        expect(birthInput).toBeInTheDocument()
        expect(usernameInput).toBeInTheDocument()
        expect(passwordInput).toBeInTheDocument()
    })

    test("should hava submit button", () => {
        render(<Cadastro/>)

        const submitButton = screen.getByRole("button")
        expect(submitButton).toBeInTheDocument();
    })

    test("should change the value of input", () => {
        render(<Cadastro/>)

        const nameInput = screen.getByPlaceholderText("Maria da Penha");
        fireEvent.change(nameInput, {
            target: {
                value: "lucas"
            }
        })
        expect(nameInput.value).toBe("lucas")
    })
})