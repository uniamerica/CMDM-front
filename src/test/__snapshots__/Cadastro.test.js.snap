// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cadastro Component should verify if the component had changes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="formCadastro"
      >
        <form>
          <div
            class="formField"
          >
            <label>
              Nome completo
            </label>
            <input
              id="name"
              name="name"
              placeholder="Maria da Penha"
              type="text"
              value=""
            />
          </div>
          <div
            class="formField"
          >
            <label>
              Email
            </label>
            <input
              id="mail"
              name="mail"
              placeholder="maria@email.com"
              type="text"
              value=""
            />
          </div>
          <div
            class="formField"
          >
            <label>
              Telefone
            </label>
            <input
              id="phone"
              name="phone"
              placeholder="(DDD) 99999-0000"
              type="text"
              value=""
            />
          </div>
          <div
            class="formField"
          >
            <label>
              Data de nascimento
            </label>
            <input
              id="birth"
              name="birth"
              placeholder="01/01/1990"
              type="date"
              value=""
            />
          </div>
          <div
            class="formField"
          >
            <label>
              Nome de usu치rio
            </label>
            <input
              id="username"
              name="username"
              placeholder="MariaPenha123"
              type="text"
              value=""
            />
          </div>
          <div
            class="formField"
          >
            <label>
              Senha
            </label>
            <input
              id="password"
              name="password"
              placeholder="************"
              type="text"
              value=""
            />
          </div>
          <button
            class="submit"
            id="cadastroSubmit"
            name="submitButton"
            type="submit"
          >
            Cadastrar-se
          </button>
          <a
            href="/login"
          >
            J치 tem cadastro?
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="formCadastro"
    >
      <form>
        <div
          class="formField"
        >
          <label>
            Nome completo
          </label>
          <input
            id="name"
            name="name"
            placeholder="Maria da Penha"
            type="text"
            value=""
          />
        </div>
        <div
          class="formField"
        >
          <label>
            Email
          </label>
          <input
            id="mail"
            name="mail"
            placeholder="maria@email.com"
            type="text"
            value=""
          />
        </div>
        <div
          class="formField"
        >
          <label>
            Telefone
          </label>
          <input
            id="phone"
            name="phone"
            placeholder="(DDD) 99999-0000"
            type="text"
            value=""
          />
        </div>
        <div
          class="formField"
        >
          <label>
            Data de nascimento
          </label>
          <input
            id="birth"
            name="birth"
            placeholder="01/01/1990"
            type="date"
            value=""
          />
        </div>
        <div
          class="formField"
        >
          <label>
            Nome de usu치rio
          </label>
          <input
            id="username"
            name="username"
            placeholder="MariaPenha123"
            type="text"
            value=""
          />
        </div>
        <div
          class="formField"
        >
          <label>
            Senha
          </label>
          <input
            id="password"
            name="password"
            placeholder="************"
            type="text"
            value=""
          />
        </div>
        <button
          class="submit"
          id="cadastroSubmit"
          name="submitButton"
          type="submit"
        >
          Cadastrar-se
        </button>
        <a
          href="/login"
        >
          J치 tem cadastro?
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
